### PR TITLE
allows bounty URLs to be blocked 

### PR DESCRIPTION
**Pull request by owocki**
##### Description

allows bounty URLs to be blocked 

##### Refers/Fixes

see #product-feedback ; this is a user request

##### Testing

tested locally

### CODE REVIEW

**Comment by ddevault** [comment]
Thanks! Seems like this is missing server-side blacklisting, though. Client side validation is the same as no validation, in practice.

**Comment by owocki** [comment]
> Thanks! Seems like this is missing server-side blacklisting, though. Client side validation is the same as no validation, in practice.

fair! let me see what i can cludge into the backend quickly.

File: app/quests/migrations/0019_auto_20191030_2103.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.2.3 on 2019-10-30 21:03
```
[the remark this answers is not in the document]
Comment (owocki) [reply app/quests/migrations/0019_auto_20191030_2103.py]: d8d9b1b07c13eee59284c65859533bd899629336

**Comment by owocki** [comment]
any feedback @danlipert ? gonna merge soon

File: app/assets/v2/js/pages/new_bounty.js [hunk 2]
```diff
@@ -462,6 +462,15 @@ $('#sync-issue').on('click', function(event) {
 });
 
 $('#issueURL').focusout(function() {
+  for (var i = 0; i <= document.blocked_urls.length; i++) {
+    var this_url_filter = document.blocked_urls[i];
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/new_bounty.js]: let / const

File: app/assets/v2/js/pages/new_bounty.js [hunk 3]
```diff
@@ -462,6 +462,15 @@ $('#sync-issue').on('click', function(event) {
 });
 
 $('#issueURL').focusout(function() {
+  for (var i = 0; i <= document.blocked_urls.length; i++) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/new_bounty.js]: let / const

File: app/assets/v2/js/pages/new_bounty.js [hunk 4]
```diff
@@ -462,6 +462,15 @@ $('#sync-issue').on('click', function(event) {
...
+    var this_url_filter = document.blocked_urls[i];
+
+    if ($('input[name=issueURL]').val().toLowerCase().indexOf(this_url_filter.toLowerCase()) != -1) {
+      alert('this repo is not bountyable at the request of the maintainer.');
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/new_bounty.js]: use `_alert` ?

File: app/assets/v2/js/pages/new_bounty.js (hunk 3 above)
Comment (thelostone-mc) [reply app/assets/v2/js/pages/new_bounty.js]: do we wanna add a ` document.blocked_urls` check  ?

**Comment by danlipert** [comment]
@owocki are you going to add the backend check in this branch?

**Comment by owocki** [comment]
@danlipert wont be able to do so before todays release.. i'd say, if we think this looks ok on the frontend, lets get it into todays release -- and i will add a note to create a hard-backend-exception as a hotfix or in next weeks release.

File: app/assets/v2/js/pages/new_bounty.js (hunk 3 above)
Comment (owocki) [reply app/assets/v2/js/pages/new_bounty.js]: > let / const

i increments; why would i want to make it a constant?

> document.blocked_urls

no.. itll always exist

File: app/assets/v2/js/pages/new_bounty.js (hunk 2 above)
Comment (owocki) [reply app/assets/v2/js/pages/new_bounty.js]: updated

**Comment by owocki** [comment]
@danlipert @thelostone-mc fc8db60e94dfe60be42ed77e686c9d59cae56550 addresses ur feedabcks

File: app/assets/v2/js/pages/new_bounty.js (hunk 3 above)
Comment (thelostone-mc) [reply app/assets/v2/js/pages/new_bounty.js]: > i increments; why would i want to make it a constant?

aka `let` to avoid hoisting it  :P

Comment (owocki) [reply app/assets/v2/js/pages/new_bounty.js]: TIL about hoisting.. committing a few now

File: app/assets/v2/js/pages/new_bounty.js (hunk 4 above)
Comment (owocki) [reply app/assets/v2/js/pages/new_bounty.js]: done

**Pull request merged**